### PR TITLE
looking for 'NotRegistered' error code

### PR DESCRIPTION
**Pull request by matzew**
Additional error code lookup, for cases the app has been uninstalled.

Once review is done, I will submit a different PR for master branch

### CODE REVIEW

**Comment by matzew** [comment]
@secondsun @danielpassos can you review/merge to 1.0.x branch ?

File: push/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/GCMPushNotificationSender.java [hunk 1]
```diff
@@ -150,17 +150,18 @@ private void cleanupInvalidRegistrationIDsForVariant(String variantID, Multicast
...
         for (int i = 0; i < results.size(); i++) {
             // use the current index to access the individual results
-            Result result = results.get(i);
+            final Result result = results.get(i);
 
-            // is there an error code that indicates an invalid regID ?
-            if (Constants.ERROR_INVALID_REGISTRATION.equals(result.getErrorCodeName())) {
+            final String errorCodeName = result.getErrorCodeName();
+            // is there an error code that indicates an invalid or not registered registration ID ?
+            if (Constants.ERROR_INVALID_REGISTRATION.equals(errorCodeName) || Constants.ERROR_NOT_REGISTERED.equals(errorCodeName)) {
```
Comment (secondsun) [inline push/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/GCMPushNotificationSender.java]: Maybe we should have a static Set of invalid error codes and make this line if (INVALID_CODES.conatins(errorCodeName)) {

Comment (matzew) [inline push/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/GCMPushNotificationSender.java]: @secondsun good idea - makes it also easier to update, based on future error codes

**Comment by matzew** [comment]
@secondsun updated

**Comment by matzew** [comment]
@secondsun @danielpassos any more comments? or can this be merged ?

**Comment by danielpassos** [comment]
:+1: Tested and seems works as expected

@secondsun Could you give it a spin?

**Comment by secondsun** [comment]
Sorry took me a while to try and get ups compiling.  I'm fighting the can't find grunt-cli demons right now.  Even though I have grunt-cli installed and it has worked in the past...

ლ(ಠ_ಠლ) Y U NO WebJar</snark>

**Comment by secondsun** [comment]
HRM. I'm still getting the duplicate  messages being sent.

**Comment by secondsun** [comment]
So on further investigation I'm getting duplicate messages BUT Google isn't sending any error token right now.  So it MAY be a problem on their end (which isn't unheard of)

**Comment by matzew** [comment]
> Sorry took me a while to try and get ups compiling. I'm fighting the can't find grunt-cli demons right now. Even though I have grunt-cli installed and it has worked in the past...

> ლ(ಠ_ಠლ) Y U NO WebJar

yes, @secondsun that's sometimes a problem we all have with the npm/node setup :-( It's awful.

Perhaps @lfryc and/or @lholmquist have ideas for improvements

**Comment by sebastienblanc** [comment]
Landed 714a244264593d7c7383f1f9f39a7d7b7b2ad3b4

**Comment by secondsun** [comment]
@matzew YUNOWebjar had a snark tag that vanished.  Was meant more as a joke ;)

Also, this morning Google WAS sending the Not Registered data correctly and the patch does what is said on the tin.  I know it is already merged but +1

**Pull request closed without merging**